### PR TITLE
Preparing Production Bundles With Docker

### PR DESCRIPTION
**Pull request by Schachte**
- Adds docker file
- Adds docker compose
- Updates express server
- Removes Redux
- Adds more config to hit external DDF

### CODE REVIEW

File: src/main/webapp/app.js [hunk 1]
```diff
@@ -27,29 +23,27 @@ const Application = ({ Routes, client }) => {
...
 
-render(Routes, createClient())
+render(Routes, createClientApollo())
 
 if (process.env.NODE_ENV !== 'production') {
   module.hot.accept('./routes', () => {
-    render(require('./routes').default, createClient())
+    render(require('./routes').default, createClientApollo())
   })
   module.hot.accept('./store', () => {
```
Comment (gjvera) [inline src/main/webapp/app.js]: do we need this now that you removed `store`?

File: .dockerignore [hunk 2]
```diff
@@ -0,0 +1,14 @@
+.git
+.project
+.vscode
+.settings
+.vs_code
+.settings
+.eslintrc*
+tsconfig*
+yarn.lock
+apollo.config.js
```
Comment (gjvera) [inline .dockerignore]: Should apollo.config be ignored in the container?

Comment (Schachte) [reply .dockerignore]: yah, don't need to pass anything into the container once the bundles are built.

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    build: .
+    environment: 
+      - NODE_ENV=production
+      - DDF_HOST_NAME=docker.for.mac.localhost
+      - DDF_PORT=8993
+      - EXPRESS_PORT=4000
+    command: [
+      "sh", 
+      "-c", 
+      "NODE_ENV=$${NODE_ENV} DDF_HOST_NAME=$${DDF_HOST_NAME} DDF_PORT=$${DDF_PORT} node bundle.middleware.js"
```
Comment (djblue) [inline docker-compose.yml]: I think you are missing the `EXPRESS_PORT` env variable.

File: src/main/webapp/intrigue-api/fetch.js [hunk 4]
```diff
@@ -22,8 +22,8 @@ const cacheBust = urlString => {
   const { query, ...rest } = url.parse(urlString)
   return url.format({
     ...rest,
-    hostname: 'localhost',
-    port: '8993',
+    hostname: process.env.DDF_HOST_NAME || 'localhost',
```
Comment (djblue) [inline src/main/webapp/intrigue-api/fetch.js]: Nice :+1:

Comment (djblue) [reply src/main/webapp/intrigue-api/fetch.js]: Although, do we just want to pass a single env variable with the ddf url?

File: src/main/webapp/intrigue-api/schema-graphql.js [hunk 5]
```diff
@@ -0,0 +1,322 @@
+const typeDefinitions = () => {
```
Comment (djblue) [inline src/main/webapp/intrigue-api/schema-graphql.js]: This doesn't need to be wrapped in a function.

File: src/main/graphql-server/server.js [hunk 6]
```diff
@@ -1,2 +1,15 @@
...
+const compression = require('compression')
+const app = express()
+const port = process.env.EXPRESS_PORT || 4000
+
+app.use(compression())
+app.use('/', router)
+app.use('/search/catalog', express.static('public'))
+
+app.listen(port, () => {
+  console.log(`Server is running on port ${port}`)
```
Comment (djblue) [inline src/main/graphql-server/server.js]: I think you need to add an eslint ignore for this line.

**Pull request merged**